server/openidc/middleware.py

```diff
@@ -40,6 +40,9 @@ def __call__(self, request):
 
         try:
             user = self.User.objects.get(username=openidc_email)
+            if not user.email:
+                user.email = openidc_email
+                user.save()
         except self.User.DoesNotExist:
             user = self.User(username=openidc_email, email=openidc_email)
             user.save()
```

Comment:
How come? It appears there is at most one db write for each possible case. Is it considered generally good practice to do transaction atomic in this type of code?

Reply:
I wouldn't say it's generally good practice no, this would create a transaction on every request in fact, which can be a bottleneck. But adding the atomic block would reduce the chance that a later regression in a query evaluation would contain this change as well and redo the user creation on every single request. In other words, the likelihood of an error per user would be reduced.

Reply:
Hmm, I'm not sure if I completely understand this. Could you show me how you'd change the code here exactly?

Reply:
Of course, sorry:

```python
        try:
            user = self.User.objects.get(username=openidc_email)
            if not user.email:
                with transaction.atomic():
                    user.email = openidc_email
                    user.save()
        except self.User.DoesNotExist:
            with transaction.atomic():
                user = self.User.objects.create(username=openidc_email, email=openidc_email)
```